lib/puppet/provider/service/systemd.rb

```diff
@@ -32,7 +32,14 @@ def disable
   end
 
   def get_start_link_count
-    Dir.glob("/etc/rc*.d/S??#{@resource[:name]}").length
```

Comment:
@kylog alright, I figured out why we need this block. When we query for something like 'ntp.service', we hit the systemd provider. (This is that weird thing where systemd manages the service because of the `service` appearing in the name, whereas `ntp` is managed by the debian provider and it's initscript.)

However, because of the sysvinit-systemd compatibility layer, services like `ntp` don't have a proper unit file (i.e, discoverable with `systemctl list-unit-files`). So we need to do the old `invoke-rc.d` stuff here, just like in the Debian provider, to figure out the status of the service.

It's pretty strange, but any service can be queried with `systemctl` in Debian 8 to get the running/stopped status, but services with no unit files depend on the old init scripts to get their enabled/disabled status.

This particular change is needed because although the resource and service name is `name.service`, systemd never stores and start scripts as `SXXname.service`. They are always `SXXname`, regardless of there being a unit file or not.

All in all, the compatibility layer is kind of a gross mess for us at this point.